src/libp2p/multiaddr.rs

```diff
@@ -339,9 +339,13 @@ impl<'a> ProtocolRef<'a> {
             _ => Vec::new(),
         };
 
-        let mut out = crate::util::leb128::encode_usize(code).collect::<Vec<_>>();
-        out.extend(extra);
-        iter::once(out.into_iter())
+        // Combine `code` and `extra`.
+        [
+            either::Left(crate::util::leb128::encode_usize(code).map(|b| either::Left([b]))),
```

Comment:
Will do the change for the iterators.
As for the "inner" usage of `either`, the elements produced by `encode_usize` are not of the same type as the one in `extra`, so this would give you a compilation error.